spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java

```diff
@@ -703,55 +577,35 @@ public void testNoInitialIdleDelayWhenRecentNotSupported() throws Exception {
 		when(store.getFolder(Mockito.any(URLName.class))).thenReturn(folder);
 		storeField.set(receiver, store);
 
-		doAnswer(new Answer<Object>() {
-
-			@Override
-			public Object answer(InvocationOnMock invocation) throws Throwable {
-				return folder;
-			}
-		}).when(receiver).getFolder();
+		doAnswer(invocation -> folder).when(receiver).getFolder();
 
 		MimeMessage mailMessage = mock(MimeMessage.class);
 		Flags flags = mock(Flags.class);
 		when(mailMessage.getFlags()).thenReturn(flags);
 		final Message[] messages = new Message[] { mailMessage };
 
 		final AtomicInteger shouldFindMessagesCounter = new AtomicInteger(2);
-		doAnswer(new Answer<Object>() {
-
-			@Override
-			public Object answer(InvocationOnMock invocation) throws Throwable {
-				/*
-				 * Return the message from first invocation of waitForMessages()
-				 * and in receive(); then return false in the next call to
-				 * waitForMessages() so we enter idle(); counter will be reset
-				 * to 1 in the mocked idle().
-				 */
-				if (shouldFindMessagesCounter.decrementAndGet() >= 0) {
-					return messages;
-				}
-				else {
-					return new Message[0];
-				}
+		doAnswer(invocation -> {
+			/*
+			 * Return the message from first invocation of waitForMessages()
+			 * and in receive(); then return false in the next call to
+			 * waitForMessages() so we enter idle(); counter will be reset
+			 * to 1 in the mocked idle().
+			 */
+			if (shouldFindMessagesCounter.decrementAndGet() >= 0) {
+				return messages;
 			}
-		}).when(receiver).searchForNewMessages();
-
-		doAnswer(new Answer<Object>() {
-
-			@Override
-			public Object answer(InvocationOnMock invocation) throws Throwable {
-				return null;
+			else {
+				return new Message[0];
 			}
-		}).when(receiver).fetchMessages(messages);
+		}).when(receiver).searchForNewMessages();
 
-		doAnswer(new Answer<Object>() {
+		doAnswer(invocation -> null).when(receiver).fetchMessages(messages);
 
-			@Override
-			public Object answer(InvocationOnMock invocation) throws Throwable {
-				Thread.sleep(5000);
-				shouldFindMessagesCounter.set(1);
-				return null;
-			}
+		doAnswer(invocation -> {
+			Thread.sleep(5000);
```

Comment:
Can we get rid of this long sleep somehow?
I understand that this is a `LongRunningIntegrationTest`, but still...
Sometime I run `testAll` locally and our CI plan for IO performs `testAll`, too. Twice.